File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
       int ones = 0, zeros = 0, ans = Integer.MIN_VALUE;
        for(int i = 0; i<s.length(); i++) {
            if(s.charAt(i) == '0') zeros++;
            else ones++;
            if(i != s.length() - 1) ans = Math.max(ans, zeros - ones);
        }
        
        return ans + ones;
    }
}

File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.py
class Solution(object):
    def maxScore(self, s):
        ones, zeros, ans = 0, 0, float('-inf')
        for i in range(len(s)):
            if s[i] == '0':
                zeros += 1
            else:
                ones += 1
            if i != len(s) - 1:
                ans = max(ans, zeros - ones)
                
        return ans + ones
        